pkg/puller/puller.go

```diff
@@ -104,235 +101,131 @@ func (p *Puller) manage(warmupTime time.Duration) {
 		cancel()
 	}()
 
-	// wait for warmup duration to complete
-	select {
-	case <-time.After(warmupTime):
-	case <-p.quit:
-		return
-	}
-
 	p.logger.Info("puller: warmup period complete, worker starting.")
 
 	for {
 		select {
 		case <-c:
-			// get all peers from kademlia
-			// iterate on entire bin at once (get all peers first)
-			// check how many intervals we synced with all of them
-			// pick the one with the most
-			// sync with that one
-
-			// if we're already syncing with this peer, make sure
-			// that we're syncing the correct bins according to depth
-			neighborhoodDepth := p.topology.NeighborhoodDepth()
-			syncRadius := p.reserveState.GetReserveState().StorageRadius
-
-			// we defer the actual start of syncing to get out of the iterator first
-			var (
-				peersToSync       []peer
-				peersToRecalc     []peer
-				peersDisconnected = make(map[string]peer)
-			)
 
 			p.syncPeersMtx.Lock()
 
```

Comment:
Does this `Lock()` has corresponding `Unlock()`?

Reply:
Nice catch!